panic in govc for commands containing an optional bool flag

Any command that contains an optional bool flag will panic when "-h" is supplied.  For example:

```
govc vm.change -h
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x13af29]

goroutine 1 [running]:
panic(0x712300, 0xc4200120b0)
       	/usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/vmware/govmomi/govc/flags.(*optionalBool).String(0xc420032178, 0x79e820, 0xc420032178)
       	/Users/cgreenwood/CNA/go/src/github.com/vmware/govmomi/govc/flags/optional_bool.go:43 +0x29
flag.isZeroValue(0xc420015500, 0x8e4bdf, 0x5, 0x9179ca)
       	/usr/local/go/src/flag/flag.go:393 +0x12d
flag.(*FlagSet).PrintDefaults.func1(0xc420015500)
       	/usr/local/go/src/flag/flag.go:467 +0x1d5
flag.(*FlagSet).VisitAll(0xc420074840, 0xc420053b60)
       	/usr/local/go/src/flag/flag.go:323 +0x67
flag.(*FlagSet).PrintDefaults(0xc420074840)
       	/usr/local/go/src/flag/flag.go:476 +0x46
flag.defaultUsage(0xc420074840)
       	/usr/local/go/src/flag/flag.go:509 +0x96
flag.(*FlagSet).usage(0xc420074840)
       	/usr/local/go/src/flag/flag.go:825 +0x55
flag.(*FlagSet).parseOne(0xc420074840, 0xc, 0xc4200da530, 0xc420032170)
       	/usr/local/go/src/flag/flag.go:870 +0xb4c
flag.(*FlagSet).Parse(0xc420074840, 0xc42000c200, 0x1, 0x1, 0xc4200ee548, 0x1)
       	/usr/local/go/src/flag/flag.go:915 +0x60
github.com/vmware/govmomi/govc/cli.Run(0xc42000c1f0, 0x2, 0x2, 0xc4200001a0)
       	/Users/cgreenwood/CNA/go/src/github.com/vmware/govmomi/govc/cli/command.go:130 +0x21a
main.main()
       	/Users/cgreenwood/CNA/go/src/github.com/vmware/govmomi/govc/main.go:69 +0x66
```

--- a/govc/flags/optional_bool.go
+++ b/govc/flags/optional_bool.go
@@ -40,7 +40,7 @@ func (b *optionalBool) Get() interface{} {
 }
 
 func (b *optionalBool) String() string {
-	if *b.val == nil {
+	if b.val == nil || *b.val == nil {
 		return "<nil>"
 	}
 	return fmt.Sprintf("%v", **b.val)

--- a/govc/flags/optional_bool_test.go
+++ b/govc/flags/optional_bool_test.go
@@ -51,6 +51,10 @@ func TestOptionalBool(t *testing.T) {
 		t.Fail()
 	}
 
+	if val == nil || *val != true {
+		t.Fail()
+	}
+
 	b.Value.Set("false")
 
 	if b.Value.String() != "false" {
@@ -58,6 +62,10 @@ func TestOptionalBool(t *testing.T) {
 	}
 
 	if b.Value.(flag.Getter).Get() != false {
+		t.Fail()
+	}
+
+	if val == nil || *val != false {
 		t.Fail()
 	}
 }

